torch/csrc/cudnn/Conv.cpp

```diff
@@ -225,6 +225,14 @@ void cudnn_convolution_forward(
   size_t workspaceSize;
   CHECK(cudnnGetConvolutionForwardWorkspaceSize(handle, idesc.desc, wdesc.desc,
       cdesc.desc, odesc.desc, fwdAlg, &workspaceSize));
+  size_t memFree;
+  size_t memTotal;
+  cudaMemGetInfo(&memFree, &memTotal);
```

Comment:
I think it doesn't have any API for that, and it doesn't keep online statistics right now. It would probably have to be extended.

Also, it would seem more reasonable to use `cudnnFindConvolutionForwardAlgorithmEx` with a limit on the workspace size, instead of always falling back to a GEMM.